### PR TITLE
feat(client): implement consumeSigninCode for POST /signinCodes/consume

### LINKED ISSUE #242: Support the new endpoint for consuming signinCodes
Over in mozilla/fxa-auth-server#1883, the auth server is implementing a new endpoint for consuming signin codes, to be called by the content server for mozilla/fxa-content-server#5060. We need a js client method to bridge the gap.

### PR DESCRIPTION
**Pull request by philbooth**
Fixes #242. Matches the endpoint being added to the auth server in mozilla/fxa-auth-server#1906.

@mozilla/fxa-devs r?

### CODE REVIEW

File: client/FxAccountClient.js [hunk 1]
```diff
@@ -1178,7 +1178,8 @@ define([
    * @param {String} phoneNumber Phone number sms will be sent to
    * @param {String} messageId Corresponding message id that will be sent
    * @param {Object} [options={}] Options
-   *   @param {String} [options.lang] lang Language that sms will be sent in
+   *   @param {String} [options.lang] Language that sms will be sent in
+   *   @param {Array} [options.features] Array of features to be enabled for the request
```
Comment (philbooth) [inline client/FxAccountClient.js]: This will be added to the auth server in mozilla/fxa-auth-server#1904.

File: client/FxAccountClient.js [hunk 2]
```diff
@@ -1239,6 +1244,28 @@ define([
   };
 
   /**
+   * Consume a signinCode.
+   *
+   * @method consumeSigninCode
+   * @param {String} signinCode The signinCode entered by the user
+   * @param {String} flowId Identifier for the current event flow
+   * @param {Number} flowBeginTime Timestamp for the flow.begin event
```
Comment (philbooth) [inline client/FxAccountClient.js]: I opted to pass the metricsContext properties inidividually here to more clearly stress that they are not optional on this endpoint. For every other endpoint they can be ditched, but that is not the case here.

File: tests/lib/signinCodes.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'intern!tdd',
+  'intern/chai!assert',
+  'tests/addons/environment'
+], function (tdd, assert, Environment) {
+
+  // These tests are intended to run against a mock auth-server. To test
```
Comment (vbudhram) [inline tests/lib/signinCodes.js]: Seems like only the `consumeSigninCode` will fail when using a real auth-server. Would it make sense to put this check in that test case only?

Comment (vbudhram) [inline tests/lib/signinCodes.js]: If you did that then you could test getting a invalid signinCode [response](https://github.com/mozilla/fxa-auth-server/blob/e0042b08de6a273bcb065a309ccd61080d4ef43e/lib/error.js#L56).

**Comment by philbooth** [comment]
Thanks for the feedback @vbudhram, especially because writing that extra test uncovered a stupid mistake I'd made in the client method.

This is just waiting on mozilla/fxa-auth-server#1906 to be merged now, if anyone out there is keen to give it a review?

**Pull request merged**